engine/src/main/java/org/terasology/physics/shapes/CollisionShape.java

```diff
@@ -26,6 +26,7 @@
     /**
      * Returns the axis-aligned bounding box ({@link AABB}) for the shape in the space represented
      * by {@code transform}.
+     *
      * @param transform The {@link Transform} pertaining to the space in which the AABB is to be calculated.
      * @return The {@link AABB} bounding the shape.
      */
```

Comment:
```
* @param transform The {@link Transform} to apply to the shape, after which the AABB is calculated.
* @return The {@link AABB} bounding the transformed shape.
```

Question that pops in my mind just by reading this javadoc: is the shape itself actually, permanently transformed or is this method meant to transform the shape only for the purpose of returning an AABB, with the shape stored in the object remaining untouched? I guess whatever the answer we need to get that across in this javadoc.

Reply:
The latter: the shape is itself not modified, it is transformed only for the purpose of returning the AABB.